From 93c036bb67bea16fc613c532c576bd7d4fc652a3 Mon Sep 17 00:00:00 2001
From: Mayank Dandwani <mayankkk@google.com>
Date: Tue, 4 Mar 2025 12:24:34 -0800
Subject: [PATCH] Allow core uids to register receiver as "android".

System components can run in any processes belonging to
core uids. If one of such components end up registering
a receiver, it will get registered as "android".

Bug: 387930030
Bug: 310632322
Test: builds
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:63fc17abcab6abae7411d63e15194873c6d99122)
Merged-In: I3fc4c9097602147e44ef77ad80defd33d8f5a2fc
Change-Id: I3fc4c9097602147e44ef77ad80defd33d8f5a2fc
---
 .../core/java/com/android/server/am/ActivityManagerService.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 4a4c87dcf686..2670c19aa69a 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -13810,7 +13810,7 @@ public class ActivityManagerService extends IActivityManager.Stub
                 Slog.w(TAG, "registerReceiverWithFeature: no app for " + caller);
                 return null;
             }
-            if (callerApp.info.uid != SYSTEM_UID
+            if (!UserHandle.isCore(callerApp.info.uid)
                     && !callerApp.getPkgList().containsKey(callerPackage)) {
                 throw new SecurityException("Given caller package " + callerPackage
                         + " is not running in process " + callerApp);
-- 
2.49.0.395.g12beb8f557-goog

